### PR TITLE
PUBDEV-5743- h2o.xgboost: Add a way to visualize h2o-xgboost trees

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-5743

### CODE REVIEW

**Comment by Pscheidl** [comment]
@michalkurka A proposal of changes based on requirements for https://github.com/h2oai/h2o-3/pull/2916

No javadoc yet, as I'll add it once we know this concept is at least on the right path.

**Comment by Pscheidl** [comment]
Tried to avoid visitor intentionally, as decision tree nodes have common props I believe can be extracted into a common predecessor. Please criticize.

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java [hunk 1]
```diff
@@ -104,7 +109,7 @@ public float getLeafValue(FVec feat, int root_id) {
         }
     }
 
-    static class Node implements Serializable {
+    public static class Node extends RegTreeNode implements Serializable {
```
Comment (sebhrusen) [inline xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java:112]: looks weird at first sight: basic extends specific?

Comment (Pscheidl) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java:112]: Oh yeah. Wanted to keep the original naming. The full name is RegTreeImpl.Node which is not that specific. But you are **right**.

Waiting for Michal, but yes I'll try to rename it if it won't break anything.

Comment (michalkurka) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java:112]: please do not rename Node, rename RegTreeNode instead if you need to

**Review by michalkurka — APPROVED** [review]

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java (hunk 1 above)
Comment (Pscheidl) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/tree/RegTreeImpl.java:112]: As the fully qualified name of "Node" is not node, but RegTreeImpl.Node (without package of course) I'd leave it as-is.

**Pull request merged**